lib/audit-logs/src/lib.rs

```diff
@@ -28,5 +28,167 @@
 pub mod pg;
 mod stream;
 
+use pg::AuditDatabaseContext;
+use serde::Deserialize;
+use serde::Serialize;
+use si_data_pg::PgError;
+use si_data_pg::PgPoolError;
+use si_events::Actor;
+use si_events::ChangeSetId;
+use si_events::ChangeSetStatus;
+use si_events::UserPk;
+use si_events::WorkspacePk;
+use strum::Display;
+use strum::EnumDiscriminants;
+use telemetry::prelude::*;
+use thiserror::Error;
+
 pub use stream::AuditLogsStream;
 pub use stream::AuditLogsStreamError;
+
+#[allow(missing_docs)]
+#[derive(Debug, Error)]
+pub enum AuditLogError {
+    #[error("pg error: {0}")]
+    Pg(#[from] PgError),
+    #[error("pg pool error: {0}")]
+    PgPool(#[from] PgPoolError),
+    #[error("serde json error: {0}")]
+    SerdeJson(#[from] serde_json::Error),
+}
+
+type Result<T> = std::result::Result<T, AuditLogError>;
+
+// FIXME(nick): delete this once accessor patterns are in place.
+// impl TryFrom<PgRow> for AuditLogRow {
+//     type Error = AuditLogError;
+//
+//     fn try_from(value: PgRow) -> std::result::Result<Self, Self::Error> {
+//         Ok(Self {
+//             actor: todo!(),
+//             kind: AuditLogKind::CreateChangeSet,
+//             entity_name: todo!(),
+//             timestamp: todo!(),
+//             change_set_id: todo!(),
+//         })
+//         let status_string: String = value.try_get("status")?;
+//         let status = ChangeSetStatus::try_from(status_string.as_str())?;
+//         Ok(Self {
+//             id: value.try_get("id")?,
+//             created_at: value.try_get("created_at")?,
+//             updated_at: value.try_get("updated_at")?,
+//             name: value.try_get("name")?,
+//             status,
+//             base_change_set_id: value.try_get("base_change_set_id")?,
+//             workspace_snapshot_address: value.try_get("workspace_snapshot_address")?,
+//             workspace_id: value.try_get("workspace_id")?,
+//             merge_requested_by_user_id: value.try_get("merge_requested_by_user_id")?,
+//             merge_requested_at: value.try_get("merge_requested_at")?,
+//             reviewed_by_user_id: value.try_get("reviewed_by_user_id")?,
+//             reviewed_at: value.try_get("reviewed_at")?,
+//         })
+//     }
+// }
+
+#[allow(missing_docs)]
+#[remain::sorted]
+#[derive(Debug, Deserialize, Serialize, Clone, PartialEq, Eq, Display, EnumDiscriminants)]
+pub enum AuditLogKind {
+    #[allow(missing_docs)]
+    AbandonChangeSet { from_status: ChangeSetStatus },
+    #[allow(missing_docs)]
+    CreateChangeSet,
+}
+
+#[allow(missing_docs)]
+#[remain::sorted]
+#[derive(Debug, Serialize, Deserialize, EnumDiscriminants)]
+#[serde(untagged, rename_all = "camelCase")]
+pub enum AuditLogMetadata {
+    #[allow(missing_docs)]
+    #[serde(rename_all = "camelCase")]
+    AbandonChangeSet { from_status: ChangeSetStatus },
+    #[allow(missing_docs)]
+    #[serde(rename_all = "camelCase")]
+    CreateChangeSet,
+}
```

Comment:
These mirror the types in si-frontend-types and si-events, but will be ultimately replaced. They were purely used for compile time checks with slight-to-no modifications.